### PR TITLE
Refactor offsets into top-level Transport function

### PR DESCRIPTION
**Pull request by ahmadia**
Also, interleaved DOFs are now used consistently in serial and parallel code.

### CODE REVIEW

File: proteus/Transport.py [hunk 1]
```diff
@@ -1493,21 +1493,14 @@ def findReferenceComponent_phi(diffusion):
...
         if comm.size() > 1:
             assert numericalFluxType != None and numericalFluxType.useWeakDirichletConditions,"You must use a numerical flux to apply weak boundary conditions for parallel runs"
-            self.offset = [0]
-            for ci in range(1,self.nc):
-                self.offset += [ci]
-            self.stride = [self.nc for ci in range(self.nc)]
-        #
+
+        self.setupInterleavedFieldStrides()
+
```
Comment (cekees) [inline proteus/Transport.py]: I don't mind pushing this into a function, but we do  need to preserve the ability to do both layouts, particularly in Transport.py.  The contiguous approach you deleted will always work, while the interleaved approach will break for  strong Dirichlet conditions and mixed trial spaces for multi-component PDEs.

Comment (ahmadia) [inline proteus/Transport.py]: Sorry, I misunderstood your comment yesterday.  I'll fix this later tonight to allow the application code to switch to a contiguous layout.

Comment (ahmadia) [inline proteus/Transport.py]: I redid the PR.

File: proteus/ADR.pyx [hunk 2]
```diff
@@ -798,21 +798,14 @@ class LevelModel(proteus.Transport.OneLevelTransport):
...
         if comm.size() > 1:
             assert numericalFluxType != None and numericalFluxType.useWeakDirichletConditions,"You must use a numerical flux to apply weak boundary conditions for parallel runs"
-            self.offset = [0]
-            for ci in range(1,self.nc):
-                self.offset += [ci]
-            self.stride = [self.nc for ci in range(self.nc)]
-        #
+
+        self.setupInterleavedFieldStrides()
+
```
Comment (cekees) [inline proteus/ADR.pyx]: just needs the name updated

**Comment by cekees** [comment]
This should work for now. At some point we should add both some logic to test whether the dof can be interleaved  and fall back to the contiguous layout if  not as well as allow the user to manually chose the layout.

**Comment by ahmadia** [comment]
Redone once more.  I think everything is fixed.

**Comment by cekees** [comment]
Looks good.

**Pull request merged**